SPECS/dracut/20overlayfs/overlayfs-mount.sh

```diff
@@ -131,11 +147,19 @@ mount_overlayfs() {
                 mount_volatile_persistent_volume $volume $overlay_mount_with_cnt
                 volume_mount_map[$volume]=$overlay_mount_with_cnt
             fi
+
+           # If 'ro' is in the mount options, use default values for upper and
+           # work directories.
+            if [[ "$overlayfs_mnt_option" == *"ro"* ]]; then
```

Comment:
Overlays support having multiple lowerdirs. They also support having no upperdir+workdir. This means that if you want to have a read-only overlay, you can just specify two lowerdirs and can omit upperdir and workdir.